src/DotNetWorker.Grpc/GrpcStartupOptions.cs

```diff
@@ -6,6 +6,12 @@ namespace Microsoft.Azure.Functions.Worker
 {
     internal class GrpcWorkerStartupOptions
     {
+        /// <summary>
+        /// Absolute URI of the grpc server including port.
+        /// Use this value to connect if not null, else use <see cref="Host"/> and <see cref="Port"/> to build the URI for connecting to the grpc server.
+        /// </summary>
+        public string? Uri { get; set; }
```

Comment:
We should remove `Host` and `Port`. Update options configuration to combine them onto this property if the explicit uri is not present, but host & port are.

Reply:
We do use the `Hos`t and `Port` [in one place](https://github.com/Azure/azure-functions-dotnet-worker/blob/e358ca5b804dc282e1c98b48ab6097caca0fc0fb/src/DotNetWorker.Grpc/GrpcWorkerClientFactory.cs#L129) when we are creating an instance of `GrpcCore.Channel`. That class does not have a constructor which takes a full URI, but there is one which takes a target string. The constructor we are calling will build the target string and call the other constructor. The targetString is built by `string.Format("{0}:{1}", host, port)` and further down the line passed to a native API. I do not see a constructor which takes a full Uri.

We could convert our `string? Uri` to `Uri` type which will allow us to get the Host and Port.  Thoughts?

Reply:
Yeah that works - I believe there are even analyzers that will tell you to use `Uri` type over `string`.

Reply:
Added the Uri type property and removed Host and Port.